src/components/activities/MedicalsActivity/MedicalsActivity.tsx

```diff
@@ -0,0 +1,409 @@
+
+  const onGridReady = (params: any) => {
+    setGridApi(params.api);
+    setGridColumnApi(params.columnApi);
+  };
+
+  const autoSizeAll = (skipHeader: boolean) => {
+    var allColumnIds: any = [];
+    gridColumnApi?.getAllColumns().forEach(function (column: any) {
+      allColumnIds.push(column.colId);
+    });
+    gridColumnApi?.autoSizeColumns(allColumnIds, skipHeader);
+  };
+
+  const renderMedicalTypesResults = (): void => {
+    switch (medicalTypeStatus) {
+      case "IDLE":
+      case "LOADING":
+        return;
+      case "SUCCESS":
+        medicalTypesOptions = medicalTypesFormatter(medicalTypeResults);
+        setOptions(medicalTypesOptions);
+    }
+  };
+
+  const renderSearchResults = (): JSX.Element | undefined => {
+    switch (searchStatus) {
+      case "IDLE":
+        return;
+
+      case "LOADING":
+        return <h3 className="medicals__loading">{t("common.searching")}</h3>;
+
+      case "SUCCESS":
+        return (
+          // <!-- USARE MATERIAL UI GRID? -->
+          <div className="medicalsGrid_main">
+            <AgGridReact
+              onGridReady={onGridReady}
+              onFirstDataRendered={() => autoSizeAll(false)}
+              defaultColDef={{ resizable: true }}
+              rowData={searchValue(medicalSearchResults)}
+              rowHeight={40}
+              pagination={true}
+              paginationAutoPageSize={true}
+              frameworkComponents={{
+                iconEditRenderer: (params: any) =>
+                  IconButton({
+                    url: "/editMedical/" + params.data.code,
+                    svgImage: iconEdit,
+                  }),
+                iconDeleteRenderer: (params: any) =>
+                  IconButton({
+                    url: "deleteMedical/" + params.data.code,
+                    svgImage: iconDelete,
+                  }),
+              }}
+            >
+              <AgGridColumn
+                headerName="Type"
+                field="type.description"
+              ></AgGridColumn>
+              <AgGridColumn headerName="Code" field="prod_code"></AgGridColumn>
+              <AgGridColumn headerName="Description" field="description" sortable={true} filter={true}></AgGridColumn>
+              <AgGridColumn headerName="PcsXPck" field="pcsperpck" maxWidth={100}></AgGridColumn>
+              <AgGridColumn headerName="Stock" field="{{inqty - outqty}}" maxWidth={100}></AgGridColumn>
+              <AgGridColumn headerName="Crit. Level" field="{{(inqty - outqty) <= minqty}}" maxWidth={100}></AgGridColumn>
+              <AgGridColumn headerName="Out of Stock" field="{{(inqty - outqty) == 0}}"  checkboxSelection={true}></AgGridColumn>
+              <AgGridColumn headerName="" cellRenderer="iconEditRenderer" maxWidth={100}></AgGridColumn>
+              <AgGridColumn headerName="" cellRenderer="iconDeleteRenderer" maxWidth={100}></AgGridColumn>
+            </AgGridReact> 
```

Comment:
In general you should reuse the implemented components inside the project in order to avoid duplicated code and logic. There already is a <Table> component inside the project that's responsive and allows to use sort and pagination. I gonna suggest you to use it in order to display the MedicalsActivity as we already done with the others section.

If you want to apply some exapansion to the <Table> component (e.g. better sort or something else) please open a new and different merge request with this.

Reply:
Hi,
Since we saw Table component has a completely different structure we opted for a classic grid, which looked more clean and compact for our purpose.
Anyway, if you prefer we reuse the table component we will work on that.

Reply:
It is better to use the existing table component.